### ISSUE TITLE
add option to convert MassActionJumps to ConstantRateJumps

### ISSUE THREAD

**Issue by isaacsas** [SciML/JumpProcesses.jl#427]
In some preliminary benchmarking for a talk I'm finding that when using `Direct` with a small number of jumps (~6), we can actually now get a ~30% speedup using `ConstantRateJump`s. This needs more investigation, but it may be that in the case of a small number of jumps, where we don't need function wrappers and use recursion for type-stability, we can now do better than `MassActionJump`s.

**Comment by isaacsas** [comment]
(This is also consistent with the performance difference between `MassActionJump`s and a hand-coded `Direct` method that evaluates propensities via a single function.)

**Comment by ChrisRackauckas** [comment]
That is rather surprising still, but interesting.

**Comment by isaacsas** [comment]
Yeah, I'm not yet convinced of this result, but the conversion would be easy to add and allow us to test this systematically from Catalyst-generated models in Benchmarks.
